From 7e6de4c4b3cd58fb528503b9daa471a300bcb2cc Mon Sep 17 00:00:00 2001
From: Ryan QIAN <b32804@freescale.com>
Date: Mon, 21 Nov 2011 14:43:20 +0800
Subject: ENGR00162681 iMX6 SDIO WIFI AR6003 failed to work on resuming

- Add keep power option to mmc pm_caps, since
power of SD can not be cut on mx6, which should
be indicated in host pm cap.

Signed-off-by: Ryan QIAN <b32804@freescale.com>
---
 drivers/mmc/host/sdhci-esdhc-imx.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/host/sdhci-esdhc-imx.c b/drivers/mmc/host/sdhci-esdhc-imx.c
index bb606bb..31c63ed 100644
--- a/drivers/mmc/host/sdhci-esdhc-imx.c
+++ b/drivers/mmc/host/sdhci-esdhc-imx.c
@@ -475,6 +475,7 @@ static int esdhc_pltfm_init(struct sdhci_host *host, struct sdhci_pltfm_data *pd
 		host->tuning_max = SDHCI_TUNE_CTRL_MAX;
 		host->tuning_step = SDHCI_TUNE_CTRL_STEP;
 		host->clk_mgr_en = true;
+		host->mmc->pm_caps = MMC_PM_KEEP_POWER;
 	}
 
 	/* disable card interrupt enable bit, and clear status bit
-- 
1.7.9.5

